game/src/main/kotlin/content/area/misthalin/lumbridge/FredTheFarmer.kt

```diff
@@ -0,0 +1,323 @@
+            npc<Neutral>("Once you've collected some wool you can spin it into balls.")
+            npc<Neutral>("Anything else?")
+            choice {
+                remindMeSpinning()
+                option<Happy>("That's all, thanks.")
+            }
+        }
+    }
+
+    fun ChoiceOption.remindMeSpinning(): Unit = option<Neutral>("Can you tell me how to spin wool?") {
+        npc<Neutral>("Don't worry it's quite simple!")
+        npc<Neutral>("The nearest Spinning Wheel can be found on the first floor of Lumbridge Castle.")
+        npc<Neutral>("To get to Lumbridge Castle just follow the road east.")
+        item("spinning_wheel_icon", 1200, "This icon denotes a Spinning Wheel on the world map.")
+        choice {
+            remindMeShearing()
+            option<Happy>("That's all, thanks.")
+        }
+    }
+
+    suspend fun Player.giveBlackWool() {
+        val give = inventory.removeToLimit("ball_of_black_wool", giveBlackBallsOfWool)
+        giveBlackBallsOfWool -= give
+        if (giveBlackBallsOfWool <= 0) {
+            giveBlackBallsOfWool = 0
+            allTheWool()
+        } else {
+            player<Neutral>("That's all I've got so far.")
+            npc<Neutral>("I need $giveBlackBallsOfWool more before I can pay you.")
+            player<Neutral>("Okay, I'll work on it.")
+        }
+    }
+
+    suspend fun Player.allTheWool() {
+        val amount = 2147483647
+        player<Happy>("That's the last of them.")
+        npc<Happy>("A pleasure doing business with you. You can shear my sheep whenever you want. You could even sell the wool to the Grand Exchange.")
+        npc<Angry>("But that's the white ones only, mind you!")
+        npc<Sad>("Anyway, I guess I'd better pay you.")
+        item("coins1000_2", 500, "Fred gives you some money and teaches you some Crafting techniques.")
+        set("sheep_shearer_miniquest", "completed")
+        // On runescape wiki = Upon completion of the quest, player may no longer use shears on the black sheep to obtain this form of wool, and any wool in your inventory or stored in a bank is removed (presumably by Fred).
+        inventory.removeToLimit("black_wool", amount)
+        inventory.removeToLimit("ball_of_black_wool", amount)
+        bank.removeToLimit("black_wool", amount)
+        bank.removeToLimit("ball_of_black_wool", amount)
+        experience.add(Skill.Crafting, 150.0)
+        inventory.add("coins", 2000)
+    }
+
+    suspend fun Player.secondReward() {
+        player<Neutral>("Hello again")
+        npc<Angry>("What are you doing on my land? Ah, it's the sheep shearer! The black wool you helped me with sold exceptionally well. Please accept this as an extra thanks.")
+        statement("The farmer hands over 1,940 coins.")
+        set("sheep_shearer_miniquest", "completed_with_second_reward")
+        inventory.add("coins", 1940)
+        player<Happy>("Thank you!")
```

Comment:
Ideally need to handle what happens if the players inventory is full
```suggestion
        if (inventory.add("coins", 1940)) {
            player<Happy>("Thank you!")
        } else {
            // TODO
        }
```